openomni/packet.py

```diff
@@ -66,6 +66,10 @@ def __init__(self, data=""):
             segment_len = min(Packet.MAX_BODY_SEGMENT_LEN,self.body_len,len(data)-14)
             self.body = data[13:(13+segment_len)]
             self.crc = ord(data[13+segment_len])
+            if (self.message_type[0].encode('hex') == '1d'):
+                self.minutes_active = ((ord(data[18]) & 0x0f) << 6) + (ord(data[19]) >> 2)
+                self.reservoir_level = round((float((((ord(data[19]) & 0x03) << 6) + (ord(data[20]) >> 2)))*50)/255, 0)
```

Comment:
Yes, I realize I'm asking a bit much.  I'm hoping to find some time to get an example of this checked in.  I'll describe it a bit more below, but will apologize in advance of not having time to really generate a detailed description.

I imagine that all the different types of messages will have their own classes; boluses will have amounts, temp basals will have amounts and duration, and some messages, like basal schedule update, will have larger, maybe structured attributes.